File: external/HDF-EOS/hdfeos/samples/SetupPoint.c
#include "hdf.h"

/* SetupPoint */

main()
{

    intn            status;
    int32           ptfid, PTid1, PTid2, PTid3;


    /*
     * We first open the HDF point file, "PointFile.hdf".  Because this file
     * does not already exist, we use the DFACC_CREATE access code in the
     * open statement.  The PTopen routine returns the point file id, ptfid,
     * which is used to identify the file in subsequent routines in the
     * library.
     */

    ptfid = PTopen("PointFile.hdf", DFACC_CREATE);


    PTid1 = PTcreate(ptfid, "Simple Point");
    PTid2 = PTcreate(ptfid, "FixedBuoy Point");
    PTid3 = PTcreate(ptfid, "FloatBuoy Point");

    /*
     * We now close the point interface with the PTdetach routine.  This step
     * is necessary to properly store the point information within the file.
     */

    PTdetach(PTid1);
    PTdetach(PTid2);
    PTdetach(PTid3);

    HEprint(stdout,0);


    /*
     * Finally, we close the swath file using the PTclose routine.  This will
     * release the point file handles established by PTopen.
     */

    PTclose(ptfid);
    HEprint(stdout,0);

    return;
}

File: external/HDF-EOS/hdfeos/samples/setuppoint.f
c
c  In this example we will (1) open an HDF file, and (2) create three point
c  interfaces within the file.
c

      program setuppoint

      integer*4          ptfid, ptid1, ptid2, ptid3, ptdetach, ptclose
      integer*4          ptopen, ptcreate
      
      integer DFACC_CREATE
      parameter (DFACC_CREATE=4)

c   
c      We first open the HDF swath point, "PointFile.hdf".  Because this
c      file does not already exist, we use the DFACC_CREATE access
c      code in the open statement.  The ehopen routine returns the point
c      file id, ptfid, which is used to identify the file in subsequent
c      routines in the library.
c
      ptfid = ptopen("PointFile.hdf", DFACC_CREATE)


      ptid1 = ptcreate(ptfid, "Simple Point")
      ptid2 = ptcreate(ptfid, "FixedBuoy Point")
      ptid3 = ptcreate(ptfid, "FloatBuoy Point")

c    
c     We now close the point interface with the ptdetach routine.  This step
c     is necessary to properly store the point information within the file.
c     

      status = ptdetach(ptid1)
      status = ptdetach(ptid2)
      status = ptdetach(ptid3)

c    
c     Finally, we close the point file using the ehclose routine.  This will
c     release the point file handles established by ptopen.
c     

      status = ptclose(ptfid)

      stop
      end



